src/services/validate.ts

```diff
@@ -0,0 +1,346 @@
+    // discard headers
+    // in every subsequent line, get the output at index 1 (which is the relative path of a dependency)
+    const dependencies = grepDependencies.stdout.toString()
+      .split('/n')
+      .slice(1)
+      .map(line => line.split('/t')[1]);
+
+    // check dependencies to ensure they all contain file://
+    const allDepsFiles = dependencies.every(dep => dep.includes('file://'));
+
+    return allDepsFiles ? SuccessResponse : {
+        success: false,
+        body: "Not all dependencies reside in the main chart.",
+        error: {
+          input: dependencies.toString(),
+          options: {
+            code: "E503",
+            exit: 1
+          }
+        }
+      };
+  }
+
+  private async findHooks(): Promise<ServiceResponse<string>> {
+    const hooks = spawnSync('grep', ['-Rilne', '"helm.sh/hook"', this.toValidate], {shell: true, encoding: "utf8"});
+
+    return hooks.stdout === "" ? SuccessResponse : {
+        success: false,
+        body: "Unsupported system Hooks are used in chart.",
+        error: {
+          input: hooks.stdout,
+          options: {
+            code: "E502",
+            exit: 1
+          }
+        }
+      };
+  }
+
+  private async findLookups(): Promise<ServiceResponse<string>> {
+    // Find any instance of "lookup" that starts with an opening parenthesis and ignore any white spaces between opening
+    // and the word itself
+    // eslint-disable-next-line no-useless-escape
+    const grepLookup = spawnSync('grep', ['-Rilne', '"\{\{-?\s*lookup"', `"${this.toValidate}"`], {shell: true, encoding: "utf8"});
+
+    if (grepLookup.stdout === "") {
+      return SuccessResponse;
+    }
+
+    return {
+      success: false,
+      body: "Helm charts use lookup functions",
+      error: {
+        input: "Lookup functions not permitted",
+        options: {
+          code: "E507",
+          exit: 1
+        }
+      }
+    }
+  }
+
+  private async findUnsupportedReleaseObject(skipReleaseService: boolean): Promise<ServiceResponse<string>> {
+    // beta guide 6.1.b) All Release objects (except .Name and .Namespace) are not supported
+    const unsupportedReleaseObjectsRegex = skipReleaseService ?
+      "'.Release.[Name|Namespace|Service]'" :
+      "'.Release.[Name|Namespace]'";
+
+    const allReleaseObjects = spawnSync('grep', ['-rn', '.Release.', this.toValidate], {shell: true, encoding: "utf8"});
+    const unsupportedReleaseObjects = spawnSync('grep', ['-vn', unsupportedReleaseObjectsRegex, this.toValidate], {
+      shell: true,
+      encoding: "utf8",
+      input: allReleaseObjects.stdout
+    });
+
+    return unsupportedReleaseObjects.stdout === "" ? SuccessResponse : {
+        success: false,
+        body: "Unsupported release objects are used in chart.",
+        error: {
+          input: unsupportedReleaseObjects.stdout,
+          options: {
+            code: "E504",
+            exit: 1
+          }
+        }
+      };
+  }
+
+  /**
```

Comment:
remove these commented code unless it is of some value to code readers.

Reply:
I think it's valuable because it gives them a reference on what is actually being executed in the background, but we can remove it 👍🏽